lib/reply.js

```diff
@@ -627,7 +643,7 @@ function onSendEnd (reply, payload) {
     return
   }
 
-  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204) {
+  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204 || statusCode === 304) {
```

Comment:
Let revert the changes on `304` and land the re-ordering.
```suggestion
  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204) {
```

Reply:
anyway, the PR is updated without this since the https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-7 states:

> A server MAY send a Content-Length header field in a [304 (Not Modified)](https://www.rfc-editor.org/rfc/rfc9110.html#status.304) response to a conditional GET request ([Section 15.4.5](https://www.rfc-editor.org/rfc/rfc9110.html#status.304)); a server MUST NOT send Content-Length in such a response unless its field value equals the decimal number of octets that would have been sent in the content of a [200 (OK)](https://www.rfc-editor.org/rfc/rfc9110.html#status.200) response to the same request